packages/core/src/controllers/ConvertController.ts

```diff
@@ -0,0 +1,664 @@
+        if (ConstantsUtil.POPULAR_TOKENS.includes(tokenInfo.symbol)) {
+          limitedTokens[tokenAddress] = tokenInfo
+        }
+
+        return limitedTokens
+      }, {})
+    state.suggestedTokens = Object.entries(res.tokens).reduce<Record<string, TokenInfo>>(
+      (limitedTokens, [tokenAddress, tokenInfo]) => {
+        if (ConstantsUtil.SUGGESTED_TOKENS.includes(tokenInfo.symbol)) {
+          limitedTokens[tokenAddress] = tokenInfo
+        }
+
+        return limitedTokens
+      },
+      {}
+    )
+    const networkToken = res.tokens[ConstantsUtil.NATIVE_TOKEN_ADDRESS]
+    this.setSourceToken(networkToken)
+
+    return state.tokens
+  },
+
+  async getAddressPrice(address: string) {
+    const existPrice = state.tokensPriceMap[address]
+    if (existPrice) {
+      return parseFloat(existPrice)
+    }
+    const prices = await ConvertApiController.getTokenPriceWithAddresses([address])
+    const price = prices[address] || '0'
+    state.tokensPriceMap[address] = price
+
+    return parseFloat(price)
+  },
+
+  async getNetworkTokenPrice() {
+    const prices = await ConvertApiController.getTokenPriceWithAddresses([
+      ConstantsUtil.NATIVE_TOKEN_ADDRESS
+    ])
+    const price = prices[ConstantsUtil.NATIVE_TOKEN_ADDRESS] || '0'
+    state.tokensPriceMap[ConstantsUtil.NATIVE_TOKEN_ADDRESS] = price
+    state.networkPrice = price
+  },
+
+  async getMyTokensWithBalance() {
+    const res = await ConvertApiController.getMyTokensWithBalance()
+
+    if (!res) {
+      return
+    }
+
+    await this.getInitialGasPrice()
+
+    const networkToken = res[ConstantsUtil.NATIVE_TOKEN_ADDRESS]
+
+    state.tokensPriceMap = Object.entries(res).reduce<Record<string, string>>(
+      (prices, [tokenAddress, tokenInfo]) => {
+        prices[tokenAddress] = tokenInfo.price
+
+        return prices
+      },
+      {}
+    )
+    state.myTokensWithBalance = res
+    state.networkBalanceInUSD = networkToken
+      ? NumberUtil.multiply(networkToken.balance, networkToken.price).toString()
+      : '0'
+  },
+
+  async getInitialGasPrice() {
+    const res = await ConvertApiController.getGasPrice()
+    const instant = res.instant
+    const value = typeof instant === 'object' ? res.instant.maxFeePerGas : instant
+    const gasFee = BigInt(value)
+    const gasLimit = BigInt(INITIAL_GAS_LIMIT)
+    const gasPrice = this.calculateGasPriceInUSD(gasLimit, gasFee)
+    state.gasPriceInUSD = gasPrice
+  },
+
+  async refreshConvertValues() {
+    const { fromAddress, toTokenDecimals, toTokenAddress } = this.getParams()
+
+    if (fromAddress && toTokenAddress && toTokenDecimals && !state.loading) {
+      const transaction = await this.getTransaction()
+      this.setTransactionDetails(transaction)
+    }
+  },
+
+  calculateGasPriceInEther(gas: bigint, gasPrice: bigint) {
+    const totalGasCostInWei = gasPrice * gas
+    const totalGasCostInEther = Number(totalGasCostInWei) / 1e18
```

Comment:
Define the values like `1e18` into different places, also, we should be sure that every calculations have been considered as big number and calculations are valid.